JoinColony/colonyCDapp
# Fix: Do not prematurely end the GetExpenditure query polling

## Description

So the main issue is that we're prematurely ending the GetExpenditure query polling on the PaymentBuilderWidget. Which means that we won't ever get newer expenditure data information.

I refactored the `useGetExpenditureData` hook be removing a redundant setTimeout because we don't actually need it. Basically, this hooks just needs to be very simple -- start or end the polling based on `shouldPoll`, and that's pretty much it. 

This hook is used in other areas of the app. I'm not entirely sure what the use case is in those other areas, but I am certain that as far as the Payment Builder Widget goes, we need  to poll for information whilst the Widget is mounted. So I ended up adding a `pollUntilUnmount` option which pretty much forces the poll to continue in the background regardless of original condition.

![payment-widget-poll](https://github.com/user-attachments/assets/440f8a9b-d3a5-4dd9-8ce2-356e9cc99e76)

## Testing

1. Create an Advanced Payment
2. On the Review step, click Confirm details
3. Verify that you can eventually reach the Funding step, without you having to reload the app/close and reopen the completed action component

Resolves #3719
